iOS/FVM/Framework/Controllers/FVMDamagedCarViewController.swift

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 Flying Vehicle Monster team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+
+public class FVMDamagedCarViewController: UIViewController {
+    @IBOutlet weak var damageSelector: FVMCarModelViewController!
+    @IBOutlet weak var rotationPromptUIImageView: UIImageView!
+    @IBOutlet weak var informationForUserTextView: UITextView!
+    @IBOutlet weak var closeButton: UIButton!
+    @IBOutlet weak var finishButton: UIButton!
+    public var jsonConfigurationData: String?
+    public var getReturningJSONData: ((String) -> Void)?
```

Comment:
Why it is optional?

Reply:
Also, this class doesn't know if it is json or not. Only parser knows about it. -> OCP

Reply:
It is optional because calling app should deliver implementation

Reply:
For me this is OK. As input we get a JSON and as output - I'm using function from API which are created also JSON

Reply:
nope, still you don't know what is a format of data. Your data is just **String**, not **JSON obj** (or whatever).  Only parser should know what type of data is that. Is the ideal world you inject parser into View Controller, AND yes it's just parser, NOT JsonParser. In ideal world, on this level of abstraction you don't know what data is that.